manifests/udev/rule.pp

```diff
@@ -0,0 +1,53 @@
+# Adds a custom udev rule
+#
+# @api public
+#
+# @see udev(7)
+#
+# @attr name [Pattern['^.+\.rules$']]
+#   The name of the udev rules to create
+#
+# @param $ensure
+#   Whether to drop a file or remove it
+#
+# @param path
+#   The path to the main systemd settings directory
+#
+# @param selinux_ignore_defaults
+#   If Puppet should ignore the default SELinux labels.
+#
+# @param restart_service
+#   Restart the managed service after setting the limits
+#
+# @param rules
+#   The literal udev rules you want to deploy
+#
+define systemd::udev::rule (
+  Array                             $rules,
+  Enum['present', 'absent', 'file'] $ensure                  = 'present',
+  Stdlib::Absolutepath              $path                    = '/etc/udev/rules.d',
+  Boolean                           $restart_service         = true,
+  Optional[Boolean]                 $selinux_ignore_defaults = false,
+) {
+  include systemd
+
+  if $name !~ Pattern['^.+\.rules$'] {
+    fail('$name must match Pattern["^.+\.rules)$"]')
+  }
```

Comment:
I wonder if `assert_type` would be more elegant.

Reply:
Does this look better?